### ISSUE TITLE
Textfields onInput not being called

### ISSUE THREAD

**Issue by OvermindDL1** [debois/elm-mdl#162]
Sometime between 7.0.0 and 7.3.0 my textfields onInput started to not be called.

I listen for the messages of onInput and update the internal model text field of it, I then clean up the message and set it back to the text field every chance I get.  However now I type a message, but when I am typing I am noticing that no onInput messages in my log are appearing now, where they used to do with every keypress.

The code of the textfield is

``` elm
                        Textfield.render
                            Mdl
                            (pidx
                                ++ [ if not_sending then
                                        0
                                     else
                                        1
                                   ]
                            )
                            mdl
                            [ Textfield.label "Type message here"
                            , Textfield.floatingLabel
                            , if not_sending then
                                Options.nop
                              else
                                Textfield.disabled
                            , if linecount <= 1 then
                                Options.nop
                              else
                                Options.many
                                    [ Textfield.textarea
                                    , Textfield.rows 4
                                    ]
                            , Textfield.value msg
                            , textfield_onInput MsgInput_Change
                            ]
```

Where `textfield_onInput` and its called function is:

``` elm
textfield_onInput : (String -> m) -> Textfield.Property m
textfield_onInput tagger =
    Options.set
        (\config ->
            { config
                | onInput =
                    Just (onInput tagger)
            }
        )

onInput : (String -> a) -> Attribute a
onInput tagger =
    Html.Events.onWithOptions "input" { stopPropagation = True, preventDefault = True } (Json.Decode.map tagger Html.Events.targetValue)
```

And this was just working early last week was the last I tested.

Can see the demonstration of the above issue at:  https://overminddl1.com/screenshots/work/TextField_onInput0.gif
![Bug example](https://overminddl1.com/screenshots/work/TextField_onInput0.gif)

**Comment by OvermindDL1** [comment]
Fixed it by changing `textfield_oninput` to:
```elm
textfield_onInput : (String -> m) -> Textfield.Property m
textfield_onInput tagger =
    Options.set
        (\config ->
            { config
                | listeners = config.listeners ++ [ onInput tagger ]
            }
        )
```

Apparently someone added a 'listeners' to the config and did not remove the old (and now unused) onInput.  So consider this a bug report that `onInput` is still on the `Textfield.Config` record when it is completely and entirely unused now.  Plus by removing it would have broken my code that I would have then fixed instead of having this bug silently for so long.  ;-)

Also, the `Textfield.on` really really badly needs a `Textfield.onWithOptions` too.

**Comment by vipentti** [comment]
`onWithOptions` has a chance of breaking certain things because currently we are using the fact that some events bubble onto the containing div so our event handlers reside on that and allow the user event handlers to reside on the `input`

**Comment by debois** [comment]
@OvermindDL1: can't you stop events you don't like by intercepting them higher up the DOM?

**Comment by OvermindDL1** [comment]
@debois I tried that, but then a rather amazing amount of elm-mdl's events stop working entirely.  I've had no issues with my custom onWithOptions setups yet.  Though I am careful to only use it on textfields, scroll events, and a few others.  Never with blur/focus or other such things.  Thankfully I will not have to deal with that when the javascript components are gone from this app, but that may be a long way away and I have little control over that.

**Comment by vipentti** [comment]
@OvermindDL1  How are your custom `onWithOptions` set up ?

**Comment by OvermindDL1** [comment]
I have a set of overridden events that I use that set preventDefault false, onInput, onScroll, couple of others.  I have elm-mdl use those by other wrappers, like:
```elm
textfield_onInput : (String -> m) -> Textfield.Property m
textfield_onInput tagger =
    Options.set
        (\config ->
            { config
                | listeners = config.listeners ++ [ onInput tagger ]
            }
        )
```
The onInput is using my custom onInput, which is:
```elm
onInput : (String -> a) -> Attribute a
onInput tagger =
    Html.Events.onWithOptions "input" { stopPropagation = True, preventDefault = True } (Json.Decode.map tagger Html.Events.targetValue)
```

Let's see it looks like I only stopPropogation and preventDefault on onInput and onClick, the others are not (I only add one when I notice it breaks with the existing site javascript).

**Comment by aforemny** [comment]
As this is quite old already, I will close here. If you would like to discuss this further, please re-open. If it is still an issue, does my fix in #333 solve this problem as well?
